win32ss/user/ntuser/winpos.c

```diff
@@ -3103,6 +3105,83 @@ IntDeferWindowPos( HDWP hdwp,
     return retvalue;
 }
 
+/* Win: xxxImeWindowPosChanged */
+static VOID FASTCALL IntImeWindowPosChanged(PSMWP psmwp)
+{
+    PWND pwnd, pwndDesktop, pwndNode;
+    PTHREADINFO pti;
+    PWINDOWLIST pWL;
+    HWND hwnd, hImeWnd, *phwnd;
+    PIMEWND pImeWnd;
+    PIMEUI pimeui;
+
+    pwndDesktop = UserGetDesktopWindow();
+    if (!pwndDesktop)
+        return;
+
+    pti = gptiCurrent;
+    pWL = IntBuildHwndList(pwndDesktop->spwndChild, IACE_LIST, pti);
+    if (!pWL)
+        return;
+
+    /* Scan the changed windows */
+    phwnd = pWL->ahwnd;
+    for (hwnd = *phwnd; hwnd != HWND_TERMINATOR; hwnd = *phwnd++)
+    {
+        if (pti->TIF_flags & TIF_INCLEANUP)
+            break;
+
+        pwnd = ValidateHwndNoErr(hwnd);
+        if (!pwnd ||
+            pwnd->head.pti != pti ||
+            pwnd->pcls->atomClassName != gpsi->atomSysClass[ICLS_IME])
+        {
+            continue;
+        }
+        /* Now pwnd is an IME window or an IME UI window of the current thread */
+
+        /* Get hImeWnd from pwnd */
+        _SEH2_TRY
+        {
+            ProbeForRead(pwnd, sizeof(IMEWND), 1);
+            pImeWnd = (PIMEWND)pwnd;
+            pimeui = pImeWnd->pimeui;
```

Comment:
Is it guaranteed pimeui will never be NULL?

Reply:
https://github.com/reactos/reactos/blob/a5aea8e537003654568ce0818dbeb07e3f8e4162/win32ss/user/user32/misc/imm.c#L970

The default IME window's `pimeui` is set at here. It's non-`NULL` if the user didn't modify.

Reply:
Added `NULL` check.

Reply:
@HBelusca Do you have any clue to solve this problem?

Reply:
```txt
(win32ss/user/ntuser/winpos.c:3156) err: Exception! 0xC0000005
```
`0xC0000005 == STATUS_ACCESS_VIOLATION`.

Reply:
The problem is fixed.